src/app/clusters/door-lock-server/door-lock-server.cpp

```diff
@@ -114,7 +114,9 @@ bool DoorLockServer::SetLockState(chip::EndpointId endpointId, DlLockState newLo
     // DlLockState::kNotFullyLocked has no appropriate event to send. Also it is unclear whether
     // it should schedule auto-relocking. So skip it here. Check for supported states explicitly
     // to handle possible enum extending in future.
-    VerifyOrReturnError(DlLockState::kLocked == newLockState || DlLockState::kUnlocked == newLockState, success);
+    VerifyOrReturnError(DlLockState::kLocked == newLockState || DlLockState::kUnlocked == newLockState ||
+                            DlLockState::kUnlatched == newLockState,
+                        success);
 
     // Send LockOperation event
     auto opType = (DlLockState::kLocked == newLockState) ? LockOperationTypeEnum::kLock : LockOperationTypeEnum::kUnlock;
```

Comment:
It might be good to document why for Unlatched we report "Unlock" here.

But more generally... I am not sure this makes sense.  Can you please walk me through who is reporting what events when for a lock that supports UBOLT for both the Unbolt and Unlock commands?  I don't see what ends up reporting the LockOperationTypeEnum::kUnlatch in this case, or the unlock after latch is released, unless I am missing something.  What I _do_ see is us reporting LockOperationTypeEnum::kUnlock while we are still trying to move to the "latch is pulled" state, which does not seem to match the spec.

Reply:
You are right, it should be `LockOperationTypeEnum::kUnlatch` here.

In general with support for `UBOLT` when performing the `UnboltDoor` command, the door lock will set the state to `kUnlocked` and we should report `LockOperationTypeEnum::kUnlock` like we're doing now during `UnlockDoor` without `UBOLT` support.

When performing `UnlockDoor` with `UBOLT` support, the door lock will move the lock to pull the latch and set the state to `kUnlatched` and we are supposed to report `LockOperationTypeEnum::kUnlatch`. Afterwards (usually a few seconds) the door lock will move slightly back to release the latch and set the state to `kUnlocked` and we report `LockOperationTypeEnum::kUnlock`.